File: src/main/resources/javagen/writeMap.java
_oprot.writeMapBegin(new TMap(TType.{{keyType}}, TType.{{valueType}}, _item.size));
for (Map.Entry<{{keyType}}, {{valueType}}> entry : _item.entrySet()) {
  {{keyType}} {{keyName}} = entry.getKey();
  {{keyWriter}}
  {{valueType}} {{valueName}} = entry.getValue();
  {{valueWriter}}
}
_oprot.writeMapEnd();

File: scrooge-generator/src/main/resources/scalagen/writeMap.scala
_oprot.writeMapBegin(new TMap(TType.{{keyWireConstType}}, TType.{{valueWireConstType}}, {{name}}.size))
{{name}}.foreach { _pair =>
  val {{keyName}} = _pair._1
  val {{valueName}} = _pair._2
{{#keyReadWriteInfo}}
  {{>writeValue}}
{{/keyReadWriteInfo}}
{{#valueReadWriteInfo}}
  {{>writeValue}}
{{/valueReadWriteInfo}}
}
_oprot.writeMapEnd()
